File: src/main/java/cn/bromine0x23/plantuml/controllers/render/SvgController.java
package cn.bromine0x23.plantuml.controllers.render;

import net.sourceforge.plantuml.FileFormat;
import org.springframework.http.MediaType;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

/**
 * 渲染PlantUML为SVG格式
 *
 * @author <a href="mailto:bromine0x23@163.com">Bromine0x23</a>
 */
@RestController
@RequestMapping({"/svg"})
public class SvgController extends AbstractRenderController {

	private static final MediaType CONTENT_TYPE = MediaType.parseMediaType("image/svg+xml");

	@Override
	protected FileFormat getFileFormat() {
		return FileFormat.SVG;
	}

	@Override
	protected MediaType getContentType() {
		return CONTENT_TYPE;
	}
}

File: src/main/kotlin/cn/bromine0x23/plantuml/controllers/SvgController.kt
package cn.bromine0x23.plantuml.controllers

import cn.bromine0x23.plantuml.services.RenderService
import cn.bromine0x23.plantuml.services.UmlService
import net.sourceforge.plantuml.FileFormat
import org.springframework.http.MediaType
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/svg")
class SvgController(
	umlService: UmlService,
	renderService: RenderService
): AbstractRenderController(umlService, renderService) {

	companion object {
		@JvmStatic
		private val CONTENT_TYPE = MediaType.parseMediaType("image/svg+xml")
	}

	override val format: FileFormat
		get() = FileFormat.SVG

	override val contentType: MediaType
		get() = CONTENT_TYPE
}